crates/iceberg/src/arrow/caching_delete_file_loader.rs

```diff
@@ -224,14 +224,12 @@ impl CachingDeleteFileLoader {
                 let (sender, receiver) = channel();
                 del_filter.insert_equality_delete(&task.file_path, receiver);
 
+                // Equality deletes intentionally have partial schemas. Schema evolution would add
+                // NULL values for missing REQUIRED columns, causing Arrow validation to fail.
                 Ok(DeleteFileContext::FreshEqDel {
-                    batch_stream: BasicDeleteFileLoader::evolve_schema(
-                        basic_delete_file_loader
-                            .parquet_to_batch_stream(&task.file_path)
-                            .await?,
-                        schema,
-                    )
-                    .await?,
+                    batch_stream: basic_delete_file_loader
```

Comment:
This is incorrect, according to [iceberg spec](https://iceberg.apache.org/spec/#equality-delete-files:~:text=If%20a%20delete,to%20null), we must do schema evolution. I think the correct approach is to fix arrow's schema?